src/libproc_macro/lib.rs

```diff
@@ -1430,24 +1435,39 @@ pub mod __internal {
 
         CURRENT_SESS.with(|p| {
             let _reset = Reset { prev: p.get() };
-            p.set((cx.parse_sess, cx.current_expansion.mark));
+
+            // No way to determine def location for a proc macro rigth now, so use call location.
+            let location = cx.current_expansion.mark.expn_info().unwrap().call_site;
+            // Opaque mark was already created by expansion, now create its transparent twin.
```

Comment:
Can you add some explanation here as to the "why" we need the transparent mark?

Reply:
Do you mean why it's transparent, or why we need a separate mark at all, or why we can't just use the call-site span literally?

Reply:
The latter, why do we need to anything special here.